app/services/dataAccess/RedisAccessObject.java

```diff
@@ -1,76 +1,180 @@
 package services.dataAccess;
 
+import com.google.protobuf.InvalidProtocolBufferException;
 import redis.clients.jedis.BinaryJedis;
-import services.dataAccess.proto.PostListProto;
-import services.dataAccess.proto.PostProto;
+import redis.clients.jedis.Pipeline;
+import redis.clients.jedis.Response;
+import services.dataAccess.proto.PostListProto.PostList;
+import services.dataAccess.proto.PostProto.Post;
+
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 /**
  * Created by erik on 23/10/16.
  */
 
 public class RedisAccessObject extends AbstractDataAccess {
+
     private BinaryJedis redisAccess;
```

Comment:
Possibly, but this operates under the assumption that it's dealing with binary arrays. We'd probably have to re-implement anyway and make RedisAccessObject another abstract if we switched to a different storage type.

However, I will look into it and resolve in a future pull request if I can successfully generalize.